tests/testCheckerMain.ml

```diff
@@ -113,7 +273,205 @@ let suite =
+          Ligo.Tezos.new_transaction ~seconds_passed:0 ~blocks_passed:0 ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let max_mintable_kit = CheckerEntrypoints.wrapper_view_burrow_max_mintable_kit ((user_addr, burrow_id), sealed_wrapper) in
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Mint_kit (burrow_id, max_mintable_kit)))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+
+          (* setup: increase the index significantly (emulate the effects of Receive_price) *)
+          let sealed_wrapper = set_last_price_in_wrapper sealed_wrapper (Some (Ligo.nat_from_literal "100_000_000n")) in
+
+          (* setup: let enough time pass so that the burrow becomes liquidatable *)
+          let blocks_passed = 191 in
+          Ligo.Tezos.new_transaction ~seconds_passed:(60 * blocks_passed) ~blocks_passed:blocks_passed ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Touch ()))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+
+          (* Mark_for_liquidation *)
+          Ligo.Tezos.new_transaction ~seconds_passed:342 ~blocks_passed:5 ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez"); (* the user themselves can mark it *)
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Mark_for_liquidation (user_addr, burrow_id)))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+
+          (* Note: I would have liked to be able to recollateralize the burrow and try
+           * out Cancel_liquidation_slice here, but to do that we need to be able to find
+           * th leaf_ptr of the slice, and currently that's hard to do. If/when we
+           * address #150 we should be able to do that. *)
+
+          (* setup: touch to start the auction *)
+          Ligo.Tezos.new_transaction ~seconds_passed:63 ~blocks_passed:1 ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Touch ()))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+
+          (* Liquidation_auction_place_bid *)
+          let min_bid = CheckerEntrypoints.wrapper_view_current_liquidation_auction_minimum_bid ((), sealed_wrapper) in
+          Ligo.Tezos.new_transaction ~seconds_passed:394 ~blocks_passed:6 ~sender:user_addr ~amount:(Ligo.tez_from_literal "0mutez");
+          let op = CheckerMain.(CheckerEntrypoint (LazyParams (Liquidation_auction_place_bid (min_bid.auction_id, min_bid.minimum_bid)))) in
+          let _ops, sealed_wrapper = CheckerMain.main (op, sealed_wrapper) in
+
+(*
+          (* FIXME: WE HAVE A PROBLEM I THINK: THE FOLLOWING RAISES AN
+           * internalError_KitSubNegative error, which I thought was impossible.
+           * Is that because of how high the index is? Is there a bug somewhere? *)
```

Comment:
Ah that's too bad, but nice catch! Interested to see what causes this.

Reply:
Yeah, I am gonna look into this today, let's see what we find. I hope it is a bug in the implementation, but I gather it might be an issue with the spec as well. I believe that to place such a big bid, `user_addr` needs to utilize some of the kit obtained as a reward for touching checker a few calls above, but that was minted out of thin air (so it's part of the circulating kit, but not part of the outstanding kit). Touching the slices at this point then tries to burn more kit than was supposed to be burnable. I'll get to the bottom of this, but I'm afraid it might be a problem with the spec minting kit as part of calling `%touch` :confused: